pkg/scheduler/factory/framework.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package factory
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+)
+
+// FrameworkConfigProducer produces a framework's configuration.
+type FrameworkConfigProducer func() (config.Plugins, []config.PluginConfig)
+
+// FrameworkConfigProducerRegistry tracks mappings from predicates/priorities to framework config producers.
+type FrameworkConfigProducerRegistry struct {
+	// maps that associate predicates/priorities with framework plugin configurations.
+	predicateToFrameworkConfigProducer map[string]FrameworkConfigProducer
+	priorityToFrameworkConfigProducer  map[string]FrameworkConfigProducer
+}
+
+// NewFrameworkConfigProducerRegistry creates a new producer registry.
+func NewFrameworkConfigProducerRegistry() *FrameworkConfigProducerRegistry {
+	return &FrameworkConfigProducerRegistry{
+		predicateToFrameworkConfigProducer: make(map[string]FrameworkConfigProducer),
+		priorityToFrameworkConfigProducer:  make(map[string]FrameworkConfigProducer),
+	}
+}
+
+func registerProducer(name string, producer FrameworkConfigProducer, producersMap map[string]FrameworkConfigProducer) {
+	validateAlgorithmNameOrDie(name)
+	producersMap[name] = producer
+}
+
+// RegisterPredicate registers a config producer for a predicate.
+func (f *FrameworkConfigProducerRegistry) RegisterPredicate(name string, producer FrameworkConfigProducer) {
+	registerProducer(name, producer, f.predicateToFrameworkConfigProducer)
+}
+
+// RegisterPriority registers a framework config producer for a priority.
+func (f *FrameworkConfigProducerRegistry) RegisterPriority(name string, producer FrameworkConfigProducer) {
+	registerProducer(name, producer, f.priorityToFrameworkConfigProducer)
+}
+
+// IsPredicateRegistered returns true if the key exist in the predicates registry
+func (f *FrameworkConfigProducerRegistry) IsPredicateRegistered(name string) bool {
```

Comment:
Do you expect these to be useful outside of `framework_test.go`? They can just be there or be inlined.

Reply:
yes, they will be used in the followup PR.